torch/optim/adadelta.py

```diff
@@ -57,13 +61,17 @@ def step(self, closure=None):
 
                 state['step'] += 1
 
-                if group['weight_decay'] != 0:
-                    grad = grad.add(group['weight_decay'], p.data)
-
                 square_avg.mul_(rho).addcmul_(1 - rho, grad, grad)
                 std = square_avg.add(eps).sqrt_()
                 delta = acc_delta.add(eps).sqrt_().div_(std).mul_(grad)
+
```

Comment:
Can't this be:
```python
if group['weight_decay'] != 0:
   p.data.add_(-group['weight_decay'], p.data)
   p.data.add_(-group['lr'], delta)
```

Reply:
`add_(scale, tensor)` is an inplace operation over the tensor that calls this routine.